databricks/koalas/window.py

```diff
@@ -1420,37 +1439,37 @@ def count(self):
         Examples
         --------
         >>> s = ks.Series([2, 2, 3, 3, 3, 4, 4, 4, 4, 5, 5])
-        >>> s.groupby(s).expanding(3).count().sort_index()  # doctest: +NORMALIZE_WHITESPACE
+        >>> s.groupby(s).expanding(3).count().sort_index()  # doctest: +SKIP
```

Comment:
skip this for supporting pandas<1.0.0 & pandas>=1.0.0 both since their behavior are different depend on pandas version.

Reply:
How is it different?

Reply:
We should match the behaviours to pandas 1.0.0 always, and make the tests working with pandas 0.2X.X.

Reply:
@ueshin 

> What's the difference, btw?

- pandas < 1.0.0
```python
>>> s = pd.Series([2, 2, 3, 3, 3, 4, 4, 4, 4, 5, 5])
>>> s.groupby(s).expanding(3).count().sort_index()
2  0     1.0
   1     2.0
3  2     1.0
   3     2.0
   4     3.0
4  5     1.0
   6     2.0
   7     3.0
   8     4.0
5  9     1.0
   10    2.0
dtype: float64
```

- pandas >= 1.0.0
```python
>>> s.groupby(s).expanding(3).count().sort_index()
2  0     NaN
   1     NaN
3  2     NaN
   3     NaN
   4     3.0
4  5     NaN
   6     NaN
   7     3.0
   8     4.0
5  9     NaN
   10    NaN
dtype: float64
```

Reply:
@HyukjinKwon @ueshin 

I think now it is time to make the purpose of "Support pandas 1.0.0" clear.

Which of the following is exactly our purpose?

1. Make the Koalas behavior match to `pandas 1.0.0` while fixing & passing existing tests
2. Make the Koalas behavior match to `pandas 1.0.0` while keeping & skipping existing tests
3. Make the Koalas behavior match to `installed pandas` while selectively keeping & skipping depends on version of pandas.

Reply:
My opinion is 3), based on what I commented at https://github.com/databricks/koalas/issues/1194#issuecomment-583063664.
We need more feedback and let me ask some people about it.

Btw, the behavior of `s.groupby(s).expanding(3).count()` with `pandas<1.0` above seems like a bug of pandas' .. ? (https://github.com/pandas-dev/pandas/pull/30923)
If so, we can just follow the new behavior?

Reply:
Hm, i thought 1. might be a better way because pandas is just a dependency in Koalas after all, and we match our behaviours to pandas, assuming the latest pandas has a corrector behaviours..

Reply:
Let me correct a little, my opinion is still basically 3), but if the behavior changes have reasonable reasons.

As for this case, the behavior with `pandas<1.0` seems like a bug, so I think we can just fix it.

Then, are there any other behavior differences?
If no, we can fix it and merge this.

If there are still behavior differences and the differences have reasonable reasons, we should discuss one-by-one.

Reply:
@HyukjinKwon @ueshin 

Thanks for the comments :)

> Then, are there any other behavior differences?

`DataFrame.info` has changed as we have discussed at https://github.com/databricks/koalas/pull/1197/files#r373335746.

I think it's a good example to talk about this.
(they just changed the behavior of printing-out, not for fixing bug)

here is an example.

- pandas < 1.0.0
```python
>>> df.info()
<class 'pandas.core.frame.DataFrame'>
Int64Index: 4 entries, 0 to 3
Data columns (total 4 columns):
A    4 non-null int64
B    3 non-null float64
C    1 non-null float64
D    4 non-null int64
dtypes: float64(2), int64(2)
```

- pandas >= 1.0.0
```
>>> df.info()
<class 'pandas.core.frame.DataFrame'>
Int64Index: 4 entries, 0 to 3
Data columns (total 4 columns):
 #   Column  Non-Null Count  Dtype
---  ------  --------------  -----
 0   A       4 non-null      int64
 1   B       3 non-null      float64
 2   C       1 non-null      float64
 3   D       4 non-null      int64
dtypes: float64(2), int64(2)
```

Reply:
And i also thought we're following 1, since if we should follow the behavior of the version of installed pandas, why just do this for < 1.0.0 and >= 1.0.0 , not for every single version of pandas?

Reply:
I meant the major version. It might contain breaking changes.

Reply:
@ueshin 

> I meant the major version. It might contain breaking changes.

Ah, got it.

Then, just let discussion about `DataFrame.info()` for now, should we check the installed major pandas version and modify it to work differently depend on that version?? 🧐